roles/puma/templates/puma.service

```
@@ -0,0 +1,17 @@
+[Unit]
+Description=Puma HTTP Server for {{ app_name }} ({{ env }})
+After=network.target
+
+[Service]
+Type=simple
+WorkingDirectory={{ release_dir }}
+ExecStart=/bin/bash -lc '{{ rvm_command }} && bundle exec puma -C {{ puma_config_file }} -e {{ env }}'
```

Here we need to add the fnm_command too.
```suggestion
ExecStart=/bin/bash -lc '{{ rvm_command }} && {{ fnm_command }} && bundle exec puma -C {{ puma_config_file }} -e {{ env }}'
```

Otherwise, the application crashes in Ubuntu distributions with the ExecJS error we know.

```
^[[A^[[A/home/deploy/consul/shared/bundle/ruby/3.1.0/gems/execjs-2.8.1/lib/execjs/runtimes.rb:58:in `autodetect': Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes. (ExecJS::RuntimeUnavailable)
	from /home/deploy/consul/shared/bundle/ruby/3.1.0/gems/execjs-2.8.1/lib/execjs.rb:5:in `<module:ExecJS>'
	from /home/deploy/consul/shared/bundle/ruby/3.1.0/gems/execjs-2.8.1/lib/execjs.rb:4:in `<top (required)>'
```

---

I found that the puma socket was never created, or maybe created and destroyed immediately as the puma task was failing. Maybe now we can wait for the socket in all distributions and not just in Ubuntu 22.04. 🤔

---

> Maybe now we can wait for the socket to exist in all distributions and not just in Ubuntu 22.04. 🤔

When running the installer, we wait on all distributions thanks to the `or not lookup("env", "CI")` condition. Although we weren't running this step on Ubuntu 22.04 when using the CI; I've just fixed that :+1:.